p1/main.cpp

```diff
@@ -0,0 +1,38 @@
+class Foo {
+    public:
+        Foo() : count(0) {}
+
+        void first(std::function<void()> printFirst) {
+            std::unique_lock<std::mutex> lock(mtx);
+            printFirst();
+            count = 1;
+            cv.notify_all();
+        }
+
+        void second(std::function<void()> printSecond) {
+            std::unique_lock<std::mutex> lock(mtx);
+            while (count != 1) {
+                cv.wait(lock);
+            }
```

Comment:
you should use the actual conditional variable feature of the conditional variable :) 
```cpp
cv.wait(lock,[count](){return count == 1;})
```